src/code/z_sram_NES.c

```diff
@@ -1,67 +1,1661 @@
+            gSaveContext.cycleSceneFlags[i].swch0 = gSaveContext.save.permanentSceneFlags[i].swch0;
+            gSaveContext.cycleSceneFlags[i].swch1 = gSaveContext.save.permanentSceneFlags[i].swch1;
+            gSaveContext.cycleSceneFlags[i].clearedRoom = gSaveContext.save.permanentSceneFlags[i].clearedRoom;
+            gSaveContext.cycleSceneFlags[i].collectible = gSaveContext.save.permanentSceneFlags[i].collectible;
+        }
+
+        for (i = 0; i < ARRAY_COUNT(gSaveContext.unk_3DD0); i++) {
+            gSaveContext.unk_3DD0[i] = 0;
+            gSaveContext.unk_3DE0[i] = 0;
+            gSaveContext.unk_3E18[i] = 0;
+            gSaveContext.unk_3E50[i] = 0;
+            gSaveContext.unk_3E88[i] = 0;
+            gSaveContext.unk_3EC0[i] = 0;
+        }
+
+        if (gSaveContext.save.isFirstCycle) {
+            gSaveContext.save.entranceIndex = 0xD800;
+            gSaveContext.save.day = 0;
+            gSaveContext.save.time = 0x3FFF;
+        } else {
+            gSaveContext.save.entranceIndex = 0x1C00;
+            gSaveContext.nextCutsceneIndex = 0;
+            gSaveContext.save.playerForm = PLAYER_FORM_HUMAN;
+        }
+    } else {
+        gSaveContext.save.entranceIndex = D_801C6A58[gSaveContext.save.owlSaveLocation];
+        if (D_801C6A58[gSaveContext.save.owlSaveLocation] == 0x84A0 && (gSaveContext.save.weekEventReg[20] & 2)) {
+            // Unconfirmed weekEventReg: "Woodfall Temple Prison Entrance raised / Water cleansed"
+            gSaveContext.save.entranceIndex = 0xCA0;
+        } else if (D_801C6A58[gSaveContext.save.owlSaveLocation] == 0x9A80 &&
+                   (gSaveContext.save.weekEventReg[33] & 0x80)) {
+            // Unconfirmed weekEventReg: "Mountain Village Unfrozen"
+            gSaveContext.save.entranceIndex = 0xAE80;
+        }
+
+        for (i = 0; i < ARRAY_COUNT(gSaveContext.cycleSceneFlags); i++) {
+            gSaveContext.cycleSceneFlags[i].chest = gSaveContext.save.permanentSceneFlags[i].chest;
+            gSaveContext.cycleSceneFlags[i].swch0 = gSaveContext.save.permanentSceneFlags[i].swch0;
+            gSaveContext.cycleSceneFlags[i].swch1 = gSaveContext.save.permanentSceneFlags[i].swch1;
+            gSaveContext.cycleSceneFlags[i].clearedRoom = gSaveContext.save.permanentSceneFlags[i].clearedRoom;
+            gSaveContext.cycleSceneFlags[i].collectible = gSaveContext.save.permanentSceneFlags[i].collectible;
+        }
+
+        if (gSaveContext.save.unk_F65) {
+            Lib_MemCpy(D_801D88A0, gSaveContext.save.scarecrowsSong, sizeof(gSaveContext.save.scarecrowsSong));
+
+            for (i = 0; i != ARRAY_COUNT(gSaveContext.save.scarecrowsSong); i++) {}
+        }
+
+        fileNum = gSaveContext.fileNum;
+        func_80147314(sramCtx, fileNum);
+    }
+}
+#else
+#pragma GLOBAL_ASM("asm/non_matchings/code/z_sram_NES/Sram_OpenSave.s")
+#endif
+
+void func_8014546C(SramContext* sramCtx) {
```

Comment:
Do you know what the relationship between this function and the next one is? Is only one used?

Reply:
I think this function handles owl saves, but I'm not sure